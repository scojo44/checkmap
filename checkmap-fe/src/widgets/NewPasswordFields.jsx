import React from 'react'
import PasswordField from './PasswordField'

export default function NewPasswordFields({register, errors, editingProfile = false}) {
  return (
    <>
      <PasswordField name="password" label="New Password (leave blank to not change)"
        {...{register, errors}}
        inputAttributes={editingProfile && {autoComplete: 'new-password'}}
        validation={!editingProfile && {required: 'Please choose a secure password'}}
      />
      <PasswordField name="confirm" label="Confirm New Password"
        {...{register, errors}}
        validation={{validate: (cp, values) => cp === values.password || 'Passwords do not match'}}
      />
    </>
  );
}
