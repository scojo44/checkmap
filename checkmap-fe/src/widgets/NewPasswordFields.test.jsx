import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import NewPasswordFields from './NewPasswordFields'

const fakeRegister = vi.fn((name, options) => {
  return {
    name,
    onChange: x => x,
    onBlur: x => x
  }
});

const fakeErrors = {
  test: {message: 'test message'}
}

describe('NewPasswordFields Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <NewPasswordFields name="password" label="Password" register={fakeRegister} errors={fakeErrors}/>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <NewPasswordFields name="password" label="Password" register={fakeRegister} errors={fakeErrors}/>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
