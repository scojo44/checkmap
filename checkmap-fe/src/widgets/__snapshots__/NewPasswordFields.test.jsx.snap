// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewPasswordFields Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="FormField"
  >
    <label
      for="password"
    >
      Password: 
    </label>
    <input
      name="password"
      type="password"
    />
  </p>
  <p
    class="FormField"
  >
    <label
      for="confirm"
    >
      Confirm Password: 
    </label>
    <input
      name="confirm"
      type="password"
    />
  </p>
</DocumentFragment>
`;
